platform/jobs/RetirementJobs.groovy

```diff
@@ -330,3 +330,109 @@ job('user-retirement-collector') {
         mailer(mailingListMap['retirement_jobs_mailing_list'])
     }
 }
+
+// ########### retirement-partner-reporter ###########
+// Defines the job for running partner reporting
+job('retirement-partner-reporter') {
+    description('Run the partner reporting job and push the results to Google Drive.')
+
+    authorization {
+        blocksInheritance(true)
+        // Only core teams have full control of the reporter.
+        List membersWithFullControl = ['edx*platform-team', 'edx*testeng', 'edx*devops']
+        membersWithFullControl.each { emp ->
+            permissionAll(emp)
+        }
+        // Other engineering teams can view.
+        List extraMembersCanView = ['edx*learner']
+        extraMembersCanView.each { emp ->
+            permission('hudson.model.Item.Read', emp)
+            permission('hudson.model.Item.Discover', emp)
+        }
+    }
+
+    // This label is configured to disallow more than one simultaneous job
+    // per instance, and generally there are always available jenkins-worker
+    // instances idling.
+    label('jenkins-worker')
+
+    // Only one of these jobs should be running at a time per environment
+    concurrentBuild(true)
+    configure { project ->
+        project / 'properties' / 'hudson.plugins.throttleconcurrents.ThrottleJobProperty' <<
+            'paramsToUseForLimit'('ENVIRONMENT')
+        project / 'properties' / 'hudson.plugins.throttleconcurrents.ThrottleJobProperty' <<
+            'limitOneJobWithMatchingParams'('true')
+    }
+
+    // keep jobs around for 30 days
+    logRotator JENKINS_PUBLIC_LOG_ROTATOR(30)
+
+    wrappers {
+        buildUserVars() /* gives us access to BUILD_USER_ID, among other things */
+        buildName('#${BUILD_NUMBER}')
```

Comment:
Not sure if there's a better build name here, maybe add the environment?

Reply:
We use just the build number as the build name for most things, but yea probably adding the edx environment (`#${BUILD_NUMBER}, ${ENV,var="ENVIRONMENT"}`) would be helpful, just like in the user-retirement-collector job.